server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -46,169 +44,191 @@
         return creationMessage.getFrom()
-            .map(draftEmailer -> draftEmailer
-                .getEmail()
-                .map(mail -> validationResultForIncorrectAddress(allowedSenders, mail))
-                .orElse(Lists.newArrayList()))
-            .orElse(Lists.newArrayList());
+                .map(draftEmailer -> draftEmailer.getEmail()
+                        .map(allowedFromMailAddresses::contains)
+                        .orElse(false))
+                .orElse(false);
     }
 
-    private List<ValidationResult> validationResultForIncorrectAddress(List<String> allowedSenders, String mail) {
-        if (!allowedSenders.contains(mail)) {
-            return Lists.newArrayList(ValidationResult.builder()
-                .message("Invalid 'from' field. Must be one of " + allowedSenders)
-                .property(MessageProperty.from.asFieldName())
-                .build());
+    
+    private MessageWithId handleOutboxMessages(CreationMessageEntry entry, MailboxSession session) throws MailboxException, MessagingException {
+        Mailbox outbox = getMailboxWithRole(session, Role.OUTBOX).orElseThrow(() -> new MailboxNotFoundException(Role.OUTBOX.serialize()));
+        if (!isRequestForSending(entry.getValue(), session)) {
+            throw new MailboxUnhandledRequestException();
         }
-        return Lists.newArrayList();
+        Function<Long, MessageId> idGenerator = uid -> generateMessageId(session, outbox, uid);
+        return createMessageInOutboxAndSend(entry, session, outbox, idGenerator);
     }
-
-    private Map<CreationMessageId, SetError> handleCreationErrors(Predicate<CreationMessage> invalidMessagesTester,
-                                                                  Function<CreationMessage, List<ValidationResult>> toValidationResults,
-                                                                  SetMessagesRequest request) {
-        return request.getCreate().entrySet().stream()
-                .filter(e -> invalidMessagesTester.test(e.getValue()))
-                .map(e -> new AbstractMap.SimpleEntry<>(e.getKey(), buildSetErrorFromValidationResult(toValidationResults.apply(e.getValue()))))
-                .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+    
+    @VisibleForTesting
+    protected MessageWithId createMessageInOutboxAndSend(CreationMessageEntry createdEntry,
+                                                           MailboxSession session,
+                                                           Mailbox outbox, Function<Long, MessageId> buildMessageIdFromUid) throws MailboxException, MessagingException {
+        CreationMessageId creationId = createdEntry.getCreationId();
+        MessageMapper messageMapper = mailboxSessionMapperFactory.createMessageMapper(session);
+        MailboxMessage newMailboxMessage = buildMailboxMessage(createdEntry, outbox);
+        messageMapper.add(outbox, newMailboxMessage);
```

Comment:
Should be messageManager here.
Cf : use of the data layer instead of the business layer...